src/basic/mkdir.h

```diff
@@ -9,21 +9,16 @@ typedef enum MkdirFlags {
 } MkdirFlags;
 
 int mkdirat_errno_wrapper(int dirfd, const char *pathname, mode_t mode);
+
 int mkdir_safe(const char *path, mode_t mode, uid_t uid, gid_t gid, MkdirFlags flags);
 int mkdir_parents(const char *path, mode_t mode);
 int mkdir_parents_safe(const char *prefix, const char *path, mode_t mode, uid_t uid, gid_t gid, MkdirFlags flags);
 int mkdir_p(const char *path, mode_t mode);
 int mkdir_p_safe(const char *prefix, const char *path, mode_t mode, uid_t uid, gid_t gid, MkdirFlags flags);
+int mkdir_p_root(const char *root, const char *p, uid_t uid, gid_t gid, mode_t m);
 
-/* mandatory access control(MAC) versions */
-int mkdir_safe_label(const char *path, mode_t mode, uid_t uid, gid_t gid, MkdirFlags flags);
-int mkdir_parents_label(const char *path, mode_t mod);
-int mkdir_p_label(const char *path, mode_t mode);
```

Comment:
The split into mkdir-label had something to do with linking (we wanted to avoid libselinux linkage?). I don't remember the details, but it made some sense at the time…

Reply:
yes it had. but it was weirdly done. the split is kept with this, but is clean now: plain mkdir is strictly in src/basic/, and selinux mkdir is strictly in src/shared: both .c *and* .h